packages/vue-app/template/App.js

```diff
@@ -295,18 +295,37 @@ export default {
     <% } /* splitChunks.layouts */ %>
     <% } /* features.layouts */ %>
     <% if (isFullStatic) { %>
+    getRouterBase() {
+      return (this.$router.options.base || '').replace(/\/+$/, '')
+    },
+    getRoutePath(route = '/') {
+      const base = this.getRouterBase()
+      if (base && route.startsWith(base)) {
+        route = route.substr(base.length)
+      }
+      return (route.replace(/\/+$/, '') || '/').split('?')[0].split('#')[0]
+    },
+    getStaticAssetsPath(route = '/') {
```

Comment:
We may refactor `urlJoin` and `encodeURI` to `getStaticAssetFile(...path)` (usage: `getStaticAssetFile (route, 'payload.js')` || `getStaticAssetFile ('manifest.js')`

Reply:
Same here, could be expose as a module util.